src/handlers.rs

```diff
@@ -393,6 +402,149 @@ where
                 maybe_await!(wallet.verify_proof(&psbt, &msg, max_confirmation_height))?;
             Ok(json!({ "spendable": spendable }))
         }
+
+        // Payjoin Logic goes here
+        SendPayjoin{
+            uri,
+        } => {
+
+            // convert the bip21 uri into a payjoin uri, and handle error if necessary 
+            let uri = payjoin::Uri::try_from(uri)
+                .map_err(|e| Error::Generic(format!("Unable to convert BIP21 into Payjoin URI: {}", e)))?;
+
```

Comment:
I imagine extracting this logic from this match statement into a stand-alone public function could be helpful. If not for export, at least for maintainability.